### PR TITLE
Only initialize Homebrew-Cask when necessary

### PR DESCRIPTION
**Pull request by ilovezfs**
Creating directories should be deferred until needed.

Fixes homebrew/homebrew-bundle#139

### CODE REVIEW

**Comment by ilovezfs** [comment]
@adidalal Just a heads up that this is the PR you were interested in.

**Comment by vitorgalvao** [comment]
Also pinging @jawshooah.

File: lib/hbc/cli.rb [hunk 1]
```diff
@@ -280,3 +280,15 @@ def _help_for(klass)
     end
   end
 end
+
+class String
```
Comment (jawshooah) [inline lib/hbc/cli.rb]: I'd rather not add needless extensions to core classes. Why not just add a method `needs_init?(command)` to `Hbc::CLI`?

Comment (ilovezfs) [inline lib/hbc/cli.rb]: You'd need to make help and null real commands.

Comment (jawshooah) [inline lib/hbc/cli.rb]: I mean add the method to `Hbc::CLI`, not `Hbc::CLI::Base` (well, I suppose both). For example:

```rb
# Hbc::CLI

def needs_init?(command)
  command.is_a? Hbc::CLI::Base && command.needs_init?
end
```

Comment (ilovezfs) [inline lib/hbc/cli.rb]: sure, but perhaps call it "should_init" instead?

Comment (jawshooah) [inline lib/hbc/cli.rb]: That's fine with me.

File: lib/hbc/cli.rb [hunk 2]
```diff
@@ -73,6 +73,10 @@ def self.require? path
     raise unless e.to_s.include? path
   end
 
+  def self.should_init?(command)
+    (command.is_a? Class) && (command < Hbc::CLI::Base) && command.needs_init?
```
Comment (ilovezfs) [inline lib/hbc/cli.rb]: @jawshooah better?

also, is_a? won't work but < does. Do you prefer < or ancestors.include?

Comment (jawshooah) [inline lib/hbc/cli.rb]: This is fine, thank you.

**Comment by adidalal** [comment]
:+1: Thanks @ilovezfs

**Comment by ilovezfs** [comment]
@adidalal You're welcome. And https://github.com/Homebrew/homebrew-bundle/issues/139#event-546022808 is now closed. Thanks for merging so quickly.

**Comment by adidalal** [comment]
Looks like this might cause https://github.com/caskroom/homebrew-cask/issues/18606 - any insight on the issue?

**Pull request merged**